playground/frontend/playground_components/lib/src/enums/complexity_factory.dart

```diff
@@ -0,0 +1,35 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import 'complexity.dart';
+
+class ComplexityFactory {
```

Comment:
Oh, I did mean a factory method right in `class Complexity`.
When modeling after `int` and `double`, they are traditionally called `parse` if they break on failure and `tryParse` if they return `null` on failure.

See https://github.com/apache/beam/blob/2955954402f8c10aecee6d4d2230847160d3fb40/playground/frontend/playground_components/lib/src/models/sdk.dart#L80